MovieCatalogCrawler/src/main/java/moviecrawler/crawlers/BasicWebCrawler.java

```diff
@@ -1,16 +1,12 @@
 package moviecrawler.crawlers;
 
 import moviecrawler.dtos.MovieDTO;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
-import org.jsoup.nodes.Element;
-import org.jsoup.select.Elements;
+import moviecrawler.strategy.SiteStrategy;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
-import java.util.stream.Collectors;
 
 public class BasicWebCrawler {
```

Comment:
Class should implement an interface to allow extension in future. Let's say we want to add a web crawler that performs a a specific logic before actually making the crawl.

Reply:
For next versions.

Reply:
https://howtodoinjava.com/best-practices/5-class-design-principles-solid-in-java/

No need to wait. The same thing will happen only the code will be ready for extension in future. Comment is to create an interface file with the method to craw in it. BasicWebCrawler implements the interface and the logic in it remains the same. Then instead of using the class we use the interface instead in the definition and create the BasicWebCrawler the same way in the same place.